src/app/services/sms/sms.factory.ts

```diff
@@ -0,0 +1,65 @@
+import Twilio from 'twilio'
+
+import FeatureManager, {
+  FeatureNames,
+  RegisteredFeature,
+} from '../../../config/feature-manager'
+
+import { sendAdminContactOtp, sendVerificationOtp } from './sms.service'
+import { TwilioConfig } from './sms.types'
+
+interface ISmsFactory {
+  sendVerificationOtp: (
+    recipient: string,
+    otp: string,
+    formId: string,
+  ) => ReturnType<typeof sendVerificationOtp>
+  sendAdminContactOtp: (
+    recipient: string,
+    otp: string,
+    userId: string,
+  ) => ReturnType<typeof sendAdminContactOtp>
+}
+
+const smsFeature = FeatureManager.get(FeatureNames.Sms)
+
+// Exported for testing.
+export const createSmsFactory = (
+  smsFeature: RegisteredFeature<FeatureNames.Sms>,
+): ISmsFactory => {
+  if (!smsFeature.isEnabled) {
+    const errorMessage = 'SMS feature must be enabled in Feature Manager first'
+    return {
+      sendAdminContactOtp: () => {
+        throw new Error(`sendAdminContactOtp: ${errorMessage}`)
+      },
+      sendVerificationOtp: () => {
+        throw new Error(`sendVerificationOtp: ${errorMessage}`)
+      },
+    }
+  }
+
+  const {
+    twilioAccountSid,
+    twilioApiKey,
+    twilioApiSecret,
+    twilioMsgSrvcSid,
+  } = smsFeature.props
+
+  const twilioClient = Twilio(twilioApiKey, twilioApiSecret, {
+    accountSid: twilioAccountSid,
+  })
+  const twilioConfig: TwilioConfig = {
+    msgSrvcSid: twilioMsgSrvcSid,
+    client: twilioClient,
+  }
+
+  return {
+    sendVerificationOtp: (recipient, otp, formId) =>
+      sendVerificationOtp(recipient, otp, formId, twilioConfig),
+    sendAdminContactOtp: (recipient, otp, userId) =>
```

Comment:
doesn't this need to be an async function? I see it being awaited on in the user.controller.ts file

Reply:
it is an async function, `SmsService#sendVerificationOtp` has a return type of `Promise<boolean>`. Just doesn't need an annotation I think

Reply:
alright, that makes sense